### PR TITLE
Give the Service port a name so that Istio selects the correct protocol

### PR DESCRIPTION
**Pull request by fibbers**
In this case MinIO exposes regular HTTP, so name it "http-minio".

See https://istio.io/docs/ops/configuration/traffic-management/protocol-selection/

This way I can make an Istio `Gateway` and `VirtualService` that point towards the created `minio-hl-svc` `Service` and expose it through my own https endpoint.

Otherwise, Istio falls back to protocol detection and modern browser try HTTP/2 which doesn't work.

### CODE REVIEW

File: pkg/resources/services/service.go [hunk 1]
```diff
@@ -28,7 +28,7 @@ import (
 
 // NewForCluster will return a new headless Kubernetes service for a MinIOInstance
 func NewForCluster(mi *miniov1beta1.MinIOInstance) *corev1.Service {
-	minioPort := corev1.ServicePort{Port: constants.MinIOPort}
+	minioPort := corev1.ServicePort{Port: constants.MinIOPort, Name: constants.MinIOServicePortName}
```
Comment (nitisht) [inline pkg/resources/services/service.go:31]: This service is a stateless service for creating the statefulset, I think what you're look for, is the service created to access MinIO. That service is outside of the scope of operator and created here https://github.com/minio/minio-operator/blob/master/examples/minioinstance-with-external-service.yaml#L10 

You can change the yaml and add the relevant port name there.

Comment (fibbers) [reply pkg/resources/services/service.go:31]: To clarify a bit, I really mean this headless Service (which I believe is created at this place):
```
❯ kubectl --namespace minio get svc
NAME           TYPE        CLUSTER-IP   EXTERNAL-IP   PORT(S)    AGE
minio-hl-svc   ClusterIP   None         <none>        9000/TCP   36h
```
If I modify it manually so the port has the name `http-minio`, I can access it via Istio from my browser (Istio has support for headless Services more recently).

However, if that's a less elegant solution/not the intended use-case and instead I should create an explicit Service, I'll do that instead :-) (I saw the previously named minioinstance-with-external-service.yaml before, but didn't use that one, since the Istio `VirtualService` already provided enough when connected to the `minio-hl-svc` Service)

Comment (nitisht) [reply pkg/resources/services/service.go:31]: I see, okay, if Istio can leverage the headless service, it makes sense to edit the name there. Can you please update the [README](https://github.com/minio/minio-operator/blob/master/README.md) with details on how to use Istio with the operator, I believe that will be helpful for many users

Comment (fibbers) [reply pkg/resources/services/service.go:31]: Ok, I added a heading in README.md and referenced a new file with Istio-specific resources that exposes the headless Service.
